Add x-exosite-token for service calling dispatcher (Issue: MUR-9171) (#10)
exosite/open_api_integration: README.md

##### Step 1: define `x-exosite-token`
In `schema`, add a attribute called `x-exosite-token`, the value needs to be a **secure token**  provided when accessing the Murano Service API.

An example (refer to this [minimalservice.yaml](./examples/minimalservice.yaml)):
```yaml
# ...
################################################################################
#                            API Information                                   #
################################################################################
# ...
x-exosite-token: "myprivatetoken"
# ...
```

Once the service swagger has been published and processed, the token will be removed and will not be accessible from Murano anymore. 
So you MUST keep it securely saved. If you forget it you can overload the current token by updating the service swagger definition.

##### Step 2: call dispatcher

Add the `authorization` header with value `<service alias> <token value>` (Example:  `authorization: minimalservice myprivatetoken`) for all calls to the Murano Service API endpoints.

- 2.1 call Event Trigger

>   base_url = https://murano-service-api.exosite.io

Author: ```suggestion
  base_url = https://service-api.hosted.exosite.io
```

>   base_url = https://murano-service-api.hosted.exosite.io

Author: ```suggestion
  base_url = https://murano-service-api.exosite.io
```

The definitions section can be used by reference objects in order re-use specification code. If elements of the spec are copied and pasted more than once, they can be set here and reused.

----

#### Define a Token to Access Dispatcher

> We allow services, which are published on Exosite Marketplace, to access [dispatcher APIs](https://pegasus-dispatcher.hosted.exosite.io).

Reviewer: ```suggestion
We allow services, which are published on Exosite Marketplace, to access [dispatcher APIs](https://pegasus-dispatcher.hosted.exosite.io) using a security Token as defined in below steps.
```

##### Step 1: define `x-exosite-token`

> In `schema`, add a attribute called `x-exosite-token`, the value can be any string,

Reviewer: ```suggestion
In `schema`, add a attribute called `x-exosite-token`, the value needs to be a **secure token**  provided when accessing the Murano Service API.
```

> it will be used as the token when you try to call dispatcher.

Reviewer: ```suggestion
```

An example (refer to this [minimalservice.yaml](./examples/minimalservice.yaml)):
```yaml
# ...
################################################################################
#                            API Information                                   #
################################################################################
# ...
x-exosite-token: "myprivatetoken"
# ...
```

> So, we defined a service called `minimalservice` and the token is `myprivatetoken`.

Reviewer: ```suggestion
Once the service swagger has been published and processed, the token will be removed and will not be accessible from Murano anymore. 
So you MUST keep it securely saved. If you forget it you can overload the current token by updating the service swagger definition.
```

##### Step 2: call dispatcher

> Add a header `authorization: minimalservice myprivatetoken` for all the following endpoints.

Reviewer: ```suggestion
Add the `authorization` header with value `<service alias> <token value>` (Example:  `authorization: minimalservice myprivatetoken`) for all calls to the Murano Service API endpoints.
```

- 2.1 call Event Trigger

>   base_url = https://pegasus-dispatcher.hosted.exosite.io

Reviewer: ```suggestion
  base_url = https://murano-service-api.hosted.exosite.io
```
@csrl  @johnlinvc  any suggestion on the public api domain?

Author: the current domain name is defined like this: [murano_ec2/development](https://github.com/exosite/murano_ec2/commit/e65470459c3d4c8749c401010c7381e4b0500373#diff-7741f37c8b6173cd1b798b99bd0cb48fR211),  should we change it to `serviceapis`? @johnlinvc  need your suggestions!!!

Reviewer: Actually it would be better to be:
> murano-service-api.exosite.io

  `GET/POST <base_url>/api/v1/trigger/:context_id/:service_alias/:event_type`

  - context_id: the solution_id
  - service_alias: which service you want to call, eg: tsdb
  - event_type: which event to call, eg: export

- 2.2 call Service Logs / Metrics

  `GET/POST <base_url>/ws/service/:service/log`

  `GET/POST <base_url>/ws/service/:service/metric/:metric`

>   If you want to use metric, please remember to define the `x-exosite-usage-metrics` along with `x-exosite-token`, if you don't define this, dispatcher will not add metrics for you.

Reviewer: ```suggestion
  If you want to use metric, please remember to define the `x-exosite-usage-metrics` along with `x-exosite-token`, if you don't define this, your connection to the metric endpoint will get rejected.
```
